source/core.py

```diff
@@ -353,7 +353,12 @@ def main():
 	speech.initialize()
 	from speech import sayAll
 	log.debug("Initializing sayAllHandler")
-	sayAll.initialize()
+	sayAll.initialize(
+		speech.speak,
+		speech.speakObject,
+		speech.getTextInfoSpeech,
+		speech.SpeakTextInfoState,
+	)

```

Comment:
Do you think this could happen from within `speech.initialize`?

Is there any chance something could depend on sayAll, and somehow initalize the speech package, neglecting to initialize the sayAll module. It would be easy to miss that requirement. It would only be a problem if core doesn't run. Which seems to be the case in addon uninstall..... right? Though, I don't see any reason an addon would need to use sayAll during uninstall.

Reply:
I agree that it makes more sense there. It also makes the dependency more obvious.